packages/cli/src/generator.ts

```diff
@@ -67,10 +66,16 @@ export function isGlobal(
     (symbol) =>
       symbol.escapedName === symbolName ||
       symbol.escapedName === `_${symbolName}`,
-    // eslint-disable-next-line @typescript-eslint/ban-types
-  ) as unknown as { parent?: Symbol };
+  );
+
+  const declarations = getDefinedArray(foundSymbol?.getDeclarations());
+  for (const declaration of declarations) {
+    if (declaration.getSourceFile().isDeclarationFile) {
```

Comment:
I'm not sure I follow the reasoning here 🤔

Reply:
If a value is declared in a declaration file (`.d.ts`) it's likely a global. In the following case for example, the value is declared locally, and the file can't be a declaration file:

```ts
// file.ts
const require = (id: string) => undefined;

require('foo');
```

But in this example, `require` is declared by `@types/node/globals.d.ts`:

```ts
// file.ts
require('foo');
```